packages/firestore/src/api/user_data_converter.ts

```diff
@@ -367,25 +155,19 @@ export class UserDataConverter {
         validatedFieldPaths.push(fieldPath);
       }
 
-      fieldMask = new FieldMask(validatedFieldPaths);
-      fieldTransforms = context.fieldTransforms.filter(transform =>
-        fieldMask.covers(transform.field)
+      return accumulator.toMergeData(
+        updateData,
+        new FieldMask(validatedFieldPaths)
       );
+    } else {
+      return accumulator.toMergeData(updateData);
     }
-    return new ParsedSetData(
-      updateData as ObjectValue,
-      fieldMask,
-      fieldTransforms
-    );
   }
 
   /** Parse update data from an update() call. */
   parseUpdateData(methodName: string, input: AnyJs): ParsedUpdateData {
-    const context = new ParseContext(
-      UserDataSource.Update,
-      methodName,
-      FieldPath.EMPTY_PATH
-    );
+    const accumulator = new ParseAccumulator(methodName, UserDataSource.Update);
+    const context = accumulator.rootContext();
```

Comment:
Is it intentional that the rest of this method did not change (to use context.addToFieldMask() and accumulator.toUpdateData())?